src/services/ThemeService.tsx

```diff
@@ -4,6 +4,42 @@ import { Theme } from "../types/Theme";
 
 const DEFAULT_URL = import.meta.env.VITE_THEME_BASE_CDN_URL;
 
+const CACHE_KEY_PREFIX = 'rcb-theme';
+const EXPIRY_DAYS = 30;
+
+const getCachedTheme = (id: string, version: string | undefined) => {
+	const cachedTheme = version ? localStorage.getItem(`${CACHE_KEY_PREFIX}_${version}_${id}`) 
+		: localStorage.getItem(`${CACHE_KEY_PREFIX}_${id}`);
+	if (!cachedTheme) return null;
+
+	const themeData = JSON.parse(cachedTheme);
+	const now = new Date();
+	const expiryDate = new Date(themeData.expiry);
+
+	if (now > expiryDate) {
```

Comment:
The current approach stores `expiry` in the cache data to represent the specific time where the cache is invalidated. This works, but let's say the developer makes a mistake of setting cache to 30 days and now wants to revert to 1 week. This will be an issue because the client side will continue to use the cache for up to 30 days.

A possible approach to circumvent this is to have `expiry` represent the time where the cache is being stored instead. Then when checking, take the `expiry` field and add on the valid period (e.g. 30 days or 1 week) before performing the check. This means that even if it was set to 30 days before, if the developer changes it to 1 week, the next time the client performs the check, it will utilise the new 1 week value.